### PR TITLE
feat: VDR support for deactivate/update/recover

### LINKED ISSUE #2423: VDR support for deactivate/update

### PR DESCRIPTION
**Pull request by fqutishat**
closes #2423

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/framework/aries/api/vdr/vdr.go [hunk 1]
```diff
@@ -35,6 +41,9 @@ type VDR interface {
 	Read(did string, opts ...resolve.Option) (*did.DocResolution, error)
 	Store(doc *did.Doc, by *[]doc.ModifiedBy) error
```
Comment (troyronda) [inline pkg/framework/aries/api/vdr/vdr.go:42]: what's store vs update?

File: pkg/framework/aries/api/vdr/vdr.go [hunk 2]
```diff
@@ -35,6 +41,9 @@ type VDR interface {
...
 	Build(keyManager kms.KeyManager, opts ...create.Option) (*did.DocResolution, error)
+	Update(did string, opts ...update.Option) error
+	Recover(did string, opts ...recovery.Option) error
```
Comment (troyronda) [inline pkg/framework/aries/api/vdr/vdr.go:45]: Recover isn't part of did-core. https://www.w3.org/TR/did-core/#method-operations

File: pkg/framework/aries/api/vdr/vdr.go [hunk 3]
```diff
@@ -35,6 +41,9 @@ type VDR interface {
 	Read(did string, opts ...resolve.Option) (*did.DocResolution, error)
 	Store(doc *did.Doc, by *[]doc.ModifiedBy) error
 	Build(keyManager kms.KeyManager, opts ...create.Option) (*did.DocResolution, error)
```
Comment (troyronda) [inline pkg/framework/aries/api/vdr/vdr.go:43]: Is this Create?

File: pkg/framework/aries/api/vdr/deactivate/option.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package deactivate implements sidetree deactivate did option
+//
```
Comment (DRK3) [inline pkg/framework/aries/api/vdr/deactivate/option.go:7]: Empty comment line

File: pkg/framework/aries/api/vdr/vdr.go (hunk 3 above)
Comment (fqutishat) [reply pkg/framework/aries/api/vdr/vdr.go:43]: yes

File: pkg/framework/aries/api/vdr/vdr.go (hunk 1 above)
Comment (fqutishat) [reply pkg/framework/aries/api/vdr/vdr.go:42]: @troyronda I will add TODO to remove Store

File: pkg/framework/aries/api/vdr/vdr.go (hunk 2 above)
Comment (sandrask) [reply pkg/framework/aries/api/vdr/vdr.go:45]: @troyronda logically it is an update but it is hard to mash update and recover. Recover is not supposed to be used lightly.

Comment (sandrask) [reply pkg/framework/aries/api/vdr/vdr.go:45]: We can omit recover from interface for now and see how/if it will be used in apps

**Pull request closed without merging**